keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactListener.kt

```diff
@@ -146,7 +146,7 @@ class ArtifactListener(
   }
 
   private fun getLatestStoredVersion(artifact: DeliveryArtifact): String? =
-    repository.artifactVersions(artifact).sortedWith(artifact.versioningStrategy.comparator).firstOrNull()
+    repository.artifactVersions(artifact, 1).firstOrNull()
```

Comment:
This was unnecessarily retrieving multiple versions when we know we only want the first one.